Automattic/wp-calypso
# Remove domain transfer old code

## Changes proposed in this Pull Request

Clean up the code no longer used after removing `domains/transfers-redesign` feature flag.

## Testing instructions

 Go to the domain transfer pages (`/domains/manage/<domain name>/transfer/<site name>`) and make sure that the redesigned transfer pages still work as described in https://github.com/Automattic/wp-calypso/pull/59150.
